lib/Serialization/ModuleFile.cpp

```diff
@@ -339,8 +346,18 @@ class ModuleFile::DeclTableInfo {
   }
 
   static internal_key_type ReadKey(const uint8_t *data, unsigned length) {
-    // TODO: Handle special names
-    return StringRef(reinterpret_cast<const char *>(data), length);
+    uint8_t kind = endian::readNext<uint8_t, little, unaligned>(data);
+    switch (kind) {
+    case (uint8_t)DeclBaseName::Kind::Normal: {
```

Comment:
Please use a mirror enum for these, like the others in SerializationFormat.h. The idea is that enums in AST aren't usually treated as being part of the module format, and if anyone changes the raw values later it would break this.